spring-integration-core/src/main/java/org/springframework/integration/endpoint/SourcePollingChannelAdapter.java

```diff
@@ -93,6 +102,32 @@ public String getComponentType() {
 	}
 
 	@Override
+	protected boolean isReceiveOnlyAdvice(Advice advice) {
+		return advice instanceof AbstractMessageSourceAdvice;
+	}
+
+	@Override
+	protected void applyReceiveOnlyAdviceChain(Collection<Advice> chain) {
+		Class<?> targetClass = AopUtils.getTargetClass(this.source);
```

Comment:
This is redundant. because `this.source` is `MessageSource` anyway and it really has `receive` method.